### PR TITLE
Service type ClusterIP - Controller port forward through CLI

### PR DESCRIPTION
**Pull request by prithviramesh**
This PR attempts to implement the recommended network setup to secure the Fission API #22

---
This change is [Reviewable](https://reviewable.io/reviews/fission/fission/431)

### CODE REVIEW

**Comment by prithviramesh** [comment]
Thanks for all your help @smruthi2187

**Comment by smruthi2187** [comment]
I'll work on this.

**Comment by smruthi2187** [comment]
Just a thought, we may not want to keep `serviceType: ClusterIP` as default. but I have it for the review just so the test run can be verified with this change. May be once the files are reviewed, I can revert it back to `LoadBalancer`.

**Comment by soamvasani** [comment]
cc @life1347

**Comment by life1347** [comment]
Review status: 0 of 15 files reviewed at latest revision, 2 unresolved discussions.

---

*[port-forward-nats.sh, line 15 at r3](https://reviewable.io:443/reviews/fission/fission/431#-L5yWuggaE-sMXZ-QD2U:-L5yWuggaE-sMXZ-QD2V:b-jlugp7) ([raw file](https://github.com/fission/fission/blob/ff91c7c7048e0ef19616efc218fc2940fb3305dc/port-forward-nats.sh#L15)):*
> ```Shell
> fi
> 
> kubectl get pods -l svc=$svc -o name --namespace default | \
> ```

Should be able to replace namespace here

---

*[charts/fission-all/values.yaml, line 6 at r3](https://reviewable.io:443/reviews/fission/fission/431#-L5yXPfoOTcyPMqdmVeE:-L5yXPfoOTcyPMqdmVeF:b8dm23g) ([raw file](https://github.com/fission/fission/blob/ff91c7c7048e0ef19616efc218fc2940fb3305dc/charts/fission-all/values.yaml#L6)):*
> ```YAML
> 
> ## Kubernetes configuration
> ## For minikube, set this to NodePort, elsewhere use LoadBalancer.
> ```

From the comment here(https://github.com/fission/fission/issues/22#issuecomment-348711770), looks like we only need to secure fission api. If we change the serviceType here then other services (i.e router, nats) will be affected as well.
Is it intentional?  Or, can we simply change the serviceType of controller in svc.yaml?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/431)*

File: charts/fission-all/templates/NOTES.txt [hunk 1]
```diff
@@ -22,6 +22,12 @@ Windows:
   $ export FISSION_URL=http://$(kubectl --namespace fission get svc controller -o=jsonpath='{..ip}')
   $ export FISSION_ROUTER=$(kubectl --namespace fission get svc router -o=jsonpath='{..ip}')
 
+{{- else if contains "ClusterIP" .Values.serviceType }}
```
Comment (life1347) [inline charts/fission-all/templates/NOTES.txt]: Please also add changes to fission-core/templates/NOTES.txt

**Comment by smruthi2187** [comment]
Review status: 0 of 20 files reviewed at latest revision, 3 unresolved discussions.

---

*[port-forward-nats.sh, line 15 at r3](https://reviewable.io:443/reviews/fission/fission/431#-L5yWuggaE-sMXZ-QD2U:-L6FyM1xfJCMeDQCX502:bb6un1j) ([raw file](https://github.com/fission/fission/blob/ff91c7c7048e0ef19616efc218fc2940fb3305dc/port-forward-nats.sh#L15)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

Should be able to replace namespace here
</blockquote></details>

Done

---

*[charts/fission-all/values.yaml, line 6 at r3](https://reviewable.io:443/reviews/fission/fission/431#-L5yXPfoOTcyPMqdmVeE:-L6FyFL1jBRZV23ruU1F:bb6un1j) ([raw file](https://github.com/fission/fission/blob/ff91c7c7048e0ef19616efc218fc2940fb3305dc/charts/fission-all/values.yaml#L6)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

From the comment here(https://github.com/fission/fission/issues/22#issuecomment-348711770), looks like we only need to secure fission api. If we change the serviceType here then other services (i.e router, nats) will be affected as well.
Is it intentional?  Or, can we simply change the serviceType of controller in svc.yaml?
</blockquote></details>

Done

---

*[charts/fission-all/templates/NOTES.txt, line 25 at r4](https://reviewable.io:443/reviews/fission/fission/431#-L5RY1F4nz6avZe-DqPl-r4-25:-L6FyI6xlTBFC9lRTQ9-:bb6un1j) ([raw file](https://github.com/fission/fission/blob/e4f9b9cb77d63e634cc0794376d66fc80b147562/charts/fission-all/templates/NOTES.txt#L25)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

Please also add changes to fission-core/templates/NOTES.txt
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/431)*

**Comment by life1347** [comment]
LGTM

**Pull request merged**